brian2/codegen/translation.py

```diff
@@ -227,13 +228,36 @@ def render_node(self, node):
                 # Do not pull out very simple expressions (including constants
                 # and numbers)
                 sympy_expr = str_to_sympy(expr)
-                if sympy.count_ops(sympy_expr, visual=False) < 2:
+                if expression_complexity(sympy_expr) < 2:
                     return expr
                 self.n += 1
                 name = '_lio_const_'+str(self.n)
                 self.optimisations[expr] = name
             return name
         else:
+            for varname, var in self.boolvars.iteritems():
+                expr_0 = word_substitute(expr, {varname: '0.0'})
+                expr_1 = '(%s)-(%s)' % (word_substitute(expr, {varname: '1.0'}), expr_0)
+                if (is_scalar_expression(expr_0, self.variables) and is_scalar_expression(expr_1, self.variables) and
+                        not has_non_float(expr, self.variables)):
+                    # we do this check here because we don't want to apply it to statements, only expressions
+                    if expression_complexity(expr)<=4:
+                        break
+                    if expr_0 not in self.optimisations:
+                        self.n += 1
+                        name_0 = '_lio_const_'+str(self.n)
+                        self.optimisations[expr_0] = name_0
+                    else:
+                        name_0 = self.optimisations[expr_0]
+                    if expr_1 not in self.optimisations:
+                        self.n += 1
+                        name_1 = '_lio_const_'+str(self.n)
+                        self.optimisations[expr_1] = name_1
+                    else:
+                        name_1 = self.optimisations[expr_1]
+                    newexpr = '({name_0}+{name_1}*{varname})'.format(name_0=name_0, name_1=name_1,
+                                                                     varname=varname)
+                    return newexpr
```

Comment:
I think we should not use `*{varname}` directly but instead `*int({varname})` (actually, it would be good to enforce this in abstract code as well but I don't have a good idea how...). It's not guaranteed that direct multiplication with a boolean works for all languages (it works for Python and C but not for Java/RenderScript, for example).